variables.tf

```
@@ -248,6 +253,29 @@ variable "encryption_at_host_enabled" {
   default = true
 }
 
+variable "entra_id_login" {
+  description = "Configuration for enabling EntraID-based login for virtual machines. Set 'enabled' to true to activate the feature and specify the list of 'principal_ids' that are allowed to log in. Note: This feature requires at least 1GB of memory, and certain SKUs like 'Standard_B1ls', 'Basic_A0', and 'Standard_A0' are not supported."
+  type = object({
+    enabled       = optional(bool)
+    principal_ids = optional(list(string))
+  })
+  default = ({
+    enabled       = false
+    principal_ids = []
+  })
+
+  validation {
+    condition     = (var.entra_id_login.enabled && (var.size != "Standard_B1ls" && var.size != "Basic_A0" && var.size != "Standard_A0")) || !var.entra_id_login.enabled
+    error_message = "The EntraID Extension must have at least 1GB of Memory. Please chosse a greater SKU."
```

The following condition is simpler to read and to maintain.

```suggestion
    condition     = !var.entra_id_login.enabled ? true : !contains(["Standard_B1ls", "Basic_A0", "Standard_A0"], var.size)
    error_message = "Entra ID Extension requires at least 1GB of memory; set var.size to an SKU that meets this requirement."
```